projects/components/form/src/form.component.spec.ts

```diff
@@ -312,6 +311,93 @@ describe('PsFormComponent', () => {
 
       expect(savebar.mode).toBe(dataSource.savebarMode);
     });
+
+    it("should call dataSource's connect() once per new dataSource", () => {
+      const fixture = TestBed.createComponent(TestDataSourceComponent);
+      const component = fixture.componentInstance;
+      expect(component).toBeDefined();
+
+      const ds1 = createDataSource();
+      component.dataSource = ds1;
+      spyOn(ds1, 'connect').and.callThrough();
+
+      fixture.detectChanges();
+
+      expect(ds1.connect).toHaveBeenCalledTimes(1);
+
+      const ds2 = createDataSource();
+      component.dataSource = ds2;
+      spyOn(ds1, 'disconnect').and.callThrough();
+      spyOn(ds2, 'connect').and.callThrough();
+
+      fixture.detectChanges();
+
+      expect(ds2.connect).toHaveBeenCalledTimes(1);
+      expect(ds1.disconnect).toHaveBeenCalledTimes(1);
+    });
+
+    it('should update errorInView$ properly', async () => {
+      const fixture = TestBed.createComponent(TestDataSourceComponent);
+      const component = fixture.componentInstance;
+      expect(component).toBeDefined();
+
+      const ds = createDataSource();
+      component.dataSource = ds;
+      let opts: IPsFormDataSourceConnectOptions;
+      spyOn(ds, 'connect').and.callFake((options: IPsFormDataSourceConnectOptions) => {
+        opts = options;
+        return ds.cdTrigger$;
+      });
+      fixture.detectChanges();
+      expect(ds.connect).toHaveBeenCalledTimes(1);
+      expect(opts).toBeDefined();
+
+      let errorInViewCheck = (value: boolean) => expect(value).toBe(false);
+
+      opts.errorInView$.subscribe(value => errorInViewCheck(value));
+
+      ds.exception = { errorObject: new Error('asdf') };
+      ds.cdTrigger$.next();
+      fixture.detectChanges();
+
+      expect(getErrorContainer(fixture)).not.toBe(null);
+
+      errorInViewCheck = (value: boolean) => expect(value).toBe(true);

```

Comment:
If errorInView$ doesnt emit, then the assert wont be executed. You could make an array and push the emitted values into the array. Then check the content of the array outside of the subscription.